pkg/util/validation/errors.go

```diff
@@ -35,8 +47,8 @@ func newLabelNameTooLongError(series []cortexpb.LabelAdapter, labelName string)
 }
 
 func newLabelValueTooLongError(series []cortexpb.LabelAdapter, labelValue string) ValidationError {
-	return &genericValidationError{
-		message: "label value too long: %.200q metric %.200q",
+	return &labelValueTooLongError{
+		message: "label value too long for metric: %.200q name %.200q",
```

Comment:
I'd suggest to remove `message` field completely, and put this message into `(*labelValueTooLongError).Error()` method (`fmt.Sprintf` call), so that the message and correct order of parameters are together.